File: Backend/PF3311.TeleRehab.Api/Models/TherapyLog.cs
namespace PF3311.Telerehab.API.Models;

public class TherapyLog
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid PatientId { get; set; }
    public Guid TherapyId { get; set; }
    public DateTime CompletedAt { get; set; } = DateTime.UtcNow;
    public bool Completed { get; set; }
    public int MoodLevel { get; set; }
    public int PainLevel { get; set; }
    public string Comment { get; set; } = string.Empty;
}

File: App2/telerehab_app/lib/models/therapy_log.dart
class TherapyLog {
  final String patientId;
  final String therapyId;
  final bool completed;
  final int moodLevel;
  final int painLevel;
  final String comment;

  TherapyLog({
    required this.patientId,
    required this.therapyId,
    required this.completed,
    required this.moodLevel,
    required this.painLevel,
    required this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      'patientId': patientId,
      'therapyId': therapyId,
      'completed': completed,
      'moodLevel': moodLevel,
      'painLevel': painLevel,
      'comment': comment,
    };
  }
}